htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/UrlUtils.java

```diff
@@ -0,0 +1,35 @@
+package io.github.wimdeblauwe.htmx.spring.boot.mvc;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+class UrlUtils {
+
+    /**
+     * Creates a target URL by prepending the context path if {@code contextRelative}
+     * is {@code true} and the URL starts with a slash ("/").
+     *
+     * @param request
+     * @param url
+     * @param contextRelative
+     * @return
+     */
+    static String createTargetUrl(HttpServletRequest request, String url, boolean contextRelative) {
+        if (contextRelative && url.startsWith("/")) {
+            // Do not apply context path to relative URLs.
+            return getContextPath(request) + url;
+        }
+        return url;
+    }
+
+    static String getContextPath(HttpServletRequest request) {
```

Comment:
What does this method do exactly? Maybe a unit test for it could make this clearer? I mostly wonder if we can't make this more efficient because now going single character by character seems inefficient to me.

Reply:
Also copied from the `RedirectView`. I'm not sure why it was there, but it looks like a protection against double slashes at the beginning of the context path.

Reply:
There was an issue in Spring Framework [SPR-16752](https://github.com/spring-projects/spring-framework/issues/21293) that was fixed by https://github.com/spring-projects/spring-framework/commit/de18d96413465700ccc3c35fb85b65606bce3450. 

It's not clear from the issue description what exactly the problem is, but if you think we don't need this validation/protection here, I can remove it. We are not doing this in other places in the code where we use the context path.

Reply:
They probably added it for good reason. I'd suggest we also keep it.

Reply:
Ok.